packages/react/src/components/Comments/FloatingComposerController.tsx

```diff
@@ -0,0 +1,78 @@
+import {
+  BlockSchema,
+  DefaultBlockSchema,
+  DefaultInlineContentSchema,
+  DefaultStyleSchema,
+  InlineContentSchema,
+  StyleSchema,
+} from "@blocknote/core";
+import { UseFloatingOptions, flip, offset, shift } from "@floating-ui/react";
+import { ComponentProps, FC, useEffect } from "react";
+
+import { useBlockNoteEditor } from "../../hooks/useBlockNoteEditor.js";
+import { useEditorSelectionBoundingBox } from "../../hooks/useEditorSelectionBoundingBox.js";
+import { useUIElementPositioning } from "../../hooks/useUIElementPositioning.js";
+import { useUIPluginState } from "../../hooks/useUIPluginState.js";
+import { FloatingComposer } from "./FloatingComposer.js";
+
+export const FloatingComposerController = <
+  B extends BlockSchema = DefaultBlockSchema,
+  I extends InlineContentSchema = DefaultInlineContentSchema,
+  S extends StyleSchema = DefaultStyleSchema
+>(props: {
+  floatingComposer?: FC<ComponentProps<typeof FloatingComposer>>;
+  floatingOptions?: Partial<UseFloatingOptions>;
+}) => {
+  const editor = useBlockNoteEditor<B, I, S>();
+
+  if (!editor.comments) {
+    throw new Error(
+      "FloatingComposerController can only be used when BlockNote editor has enabled comments"
+    );
+  }
+
+  const comments = editor.comments;
+
+  const state = useUIPluginState(comments.onUpdate.bind(comments));
+
+  // the reference position is updated automatically when the selection moves,
+  // this can happen when the document is updated by a remote user
+  const referencePos = useEditorSelectionBoundingBox(state?.pendingComment);
+
+  useEffect(() => {
+    editor.ForceSelectionVisible = !!state?.pendingComment;
+  }, [editor, state?.pendingComment]);
```

Comment:
This is weird. Can we describe why this is needed, it also feels like it should not be handled by React, because it is being derived from the editor state?